### ISSUE TITLE
Path.compare() returns wrong result if argument is empty path

### ISSUE THREAD

**Issue by iconexperience** [paperjs/paper.js#1208]
Here is a very simple example to demonstrate the issue:

```javascript
console.log(new paper.Path.Rectangle(0, 0, 100, 100).compare(new paper.Path()));
```

The output is `4`, while correct return value would be `false`.

Fixing this is very easy. In <a href="https://github.com/paperjs/paper.js/blob/develop/src/path/Path.js#L1561">Path.js Line 1561</a> replace

`return length1 ^ length2;`

with

`return length1 == length2;`

**Comment by lehni** [comment]
This seems correct. I really need to implement unit tests for #compare(), though...

**Comment by lehni** [comment]
I just implemented some tests in c12f7c4a64e1df1688da690f41eeb82e27531b55. More detailed ones are needed...
